check/caller.go

```diff
@@ -71,8 +71,12 @@ func (vc *callerImpl) Call(r *VersionCheckRequest) (*VersionCheckResponse, error
 }
 
 func validateResponse(r *VersionCheckResponse) error {
-	if r.Current.Version == "" || r.Recommended.Version == "" {
-		return errors.New("invalid response: missing current or recommended version")
+	if r.Products == nil || len(r.Products) == 0 {
```

Comment:
`r.Products == nil` is unnecessary (`len(r.Products)` works on nil and returns 0)

Reply:
good to know, thanks